GVFS/GVFS.Common/Cleanup/GitCleanupQueue.cs

```diff
@@ -97,40 +53,30 @@ private void RunQueue()
         {
             while (true)
             {
-                lock (this.threadLock)
-                {
-                    if (this.queue.Count == 0
-                        || !this.EnlistmentRootReady())
-                    {
-                        this.thread = null;
-                        return;
-                    }
-                }
-
-                lock (this.currentStepLock)
+                try
                 {
-                    this.queue.TryDequeue(out this.currentStep);
+                    this.queue.TryTake(out this.currentStep, Timeout.Infinite, this.cancellationToken.Token);
                 }
-
-                if (this.stopping || this.currentStep == null)
+                catch (OperationCanceledException)
                 {
-                    this.thread = null;
+                    // Only gets called when stop is requested
                     return;
                 }
 
-                try
-                {
-                    this.currentStep.Execute();
-                }
-                catch (Exception e)
+                if (this.EnlistmentRootReady())
                 {
-                    this.LogErrorAndExit(
-                        telemetryKey: nameof(GitCleanupQueue),
-                        methodName: nameof(this.RunQueue),
-                        exception: e);
+                    try
+                    {
+                        this.currentStep?.Execute();

```

Comment:
Can this actually ever be null? or are you just being defensive...

Reply:
The old code was defensive so I'm being defensive too :-)

Reply:
Looking at the code, I can take it out.  I'll do that :-)

Reply:
It can be `null` if no step was ever run, right?